### ISSUE TITLE
LineReceiver.clearLineBuffer should return the line buffer which was cleared

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3573]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3573|
|Type|enhancement|
|Created|2008-12-10 22:15:25Z|
|Branch|https://github.com/twisted/twisted/tree/return-line-buffer-3573|

It's useful to be able to get the buffer being discarded.  The data might be part of the next message and parsed by a different object.  For example, a pipelined HTTP response: part of a second response might be delivered along with the end of the first response; if each response is parsed by a different object, those bytes need to be extracted from the buffer and passed elsewhere.

<details><summary>Searchable metadata</summary>

```
trac-id__3573 3573
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_return_line_buffer_3573 branches/return-line-buffer-3573
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1228947325000000 1228947325000000
changetime__1229294482000000 1229294482000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25632]) Branching to 'return-line-buffer-3573'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [25633]) Make LineReceiver.clearLineBuffer return the cleared buffer

refs [#3573](https://github.com/twisted/twisted/issues/3573)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Added the one and only unit test for `LineReceiver.clearLineBuffer` and added the desired behavior.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Cool, +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [25664]) Merge return-line-buffer-3573

Author: exarkun, itamarst
Reviewer: therve
Fixes: [#3573](https://github.com/twisted/twisted/issues/3573)

Change `LineReceiver.clearLineBuffer` so that it returns the bytes from the line
buffer which it clears.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
